packages/inter-protocol/src/auction/scheduler.js

```diff
@@ -182,65 +229,84 @@ export const makeScheduler = async (
 
       void E(timer).cancel(stepCancelToken);
     }
+
+    if (schedule) {
+      publishSchedule(time);
+    }
   };
 
-  const scheduleRound = time => {
-    trace('nextRound', time);
+  // schedule the wakeups for the steps of this round
+  const scheduleSteps = now => {
     if (!liveSchedule) throw Fail`liveSchedule not defined`;
-    assert(liveSchedule);
 
     const { startTime } = liveSchedule;
     trace('START ', startTime);
 
-    const startDelay =
-      TimeMath.compareAbs(startTime, time) > 0
-        ? TimeMath.subtractAbsAbs(startTime, time)
+    const delayFromNow =
+      TimeMath.compareAbs(startTime, now) > 0
+        ? TimeMath.subtractAbsAbs(startTime, now)
         : TimeMath.subtractAbsAbs(startTime, startTime);
 
+    trace('repeating', now, delayFromNow, startTime);
+
     void E(timer).repeatAfter(
-      startDelay,
+      delayFromNow,
       liveSchedule.clockStep,
       Far('SchedulerWaker', {
         wake(t) {
+          trace('wake step', t);
           clockTick(t, liveSchedule);
         },
       }),
       stepCancelToken,
     );
   };
 
+  // schedule a wakeup for the next round
   const scheduleNextRound = start => {
-    trace(`SCHED   nextRound`, start);
+    trace(`nextRound`, start);
     void E(timer).setWakeup(
       start,
       Far('SchedulerWaker', {
         wake(time) {
           // eslint-disable-next-line no-use-before-define
           void startAuction(time);
+          publishSchedule(time);
         },
       }),
     );
   };
 
-  const startAuction = async time => {
+  const startAuction = async now => {
     !liveSchedule || Fail`can't start an auction round while one is active`;
 
+    assert(nextSchedule);
     liveSchedule = nextSchedule;
+
     const after = TimeMath.addAbsRel(
-      // @ts-expect-error guarded three lines up.
       liveSchedule.startTime,
       TimeMath.toRel(1n, timerBrand),
     );
     nextSchedule = computeRoundTiming(params, after);
-    scheduleRound(time);
-    scheduleNextRound(nextSchedule.startTime);
+
+    scheduleSteps(now);
+    scheduleNextRound(
+      TimeMath.subtractAbsRel(nextSchedule.startTime, nextSchedule.startDelay),
+    );
   };
 
-  const baseNow = await E(timer).getCurrentTimestamp();
   // XXX manualTimer returns a bigint, not a timeRecord.
-  const now = TimeMath.toAbs(baseNow, timerBrand);
-  nextSchedule = computeRoundTiming(params, now);
-  scheduleNextRound(nextSchedule.startTime);
+  const baseNow = await E(timer).getCurrentTimestamp();
+  nextSchedule = computeRoundTiming(
+    params,
+    TimeMath.toAbs(baseNow, timerBrand),
+  );
+  const firstStart = TimeMath.subtractAbsRel(
```

Comment:
this looks to be different from `baseNow`. why is `publishSchedule = now => {` taking an argument that isn't *now*?

Reply:
Nice. So now `publishSchedule` is called three places:
1. end of `clockTick`
2. after this scheduleNextRound()
3. after wake's `startAuction` which is the one other call to `scheduleNextRound()`.

Looking through the cases I think this can be simplified to two sites:
1. end of `clockTick`
2. end of `scheduleNextRound`

That makes more sense to me. Documenting that on `publishSchedule` would help too:
"To be called whenever the clock ticks or a next round is scheduled"

Reply:
done